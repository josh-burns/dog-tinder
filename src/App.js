import "./App.css";
import Card from "./components/Card";
import React, { useState, useLayoutEffect } from "react";
import axios from "axios";

import styled from "styled-components";

const AppStyle = styled.div`
  width: 300px;
  position: absolute;
  top: 50%;
  left: 50%;
  right: 50%;
  /* -ms-transform: translateY(-50%); */
  transform: translateY(-50%);
`;

export default function App() {
  const [dogs, setdogs] = useState([]);

  // TODO: get 10 images and save state of objects

  async function dataFetch() {
    axios.get(`https://dog.ceo/api/breeds/image/random`).then((res) => {
      const dogImage = res.data.message;
      let arr = [dogImage];
      console.log({ arr });
      setdogs((dogs) => [...dogs, arr]);
    });
  }

  useLayoutEffect(() => {
    for (let i = 0; i < 50; i++) {
      dataFetch();
      console.log({ dogs });
    }
  }, []);

  return (
    <AppStyle>
      <Card dataObject={dogs}> This is a Card</Card>
    </AppStyle>
  );
}
