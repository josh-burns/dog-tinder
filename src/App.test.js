import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import App from "./App";

test("renders app component", () => {
  render(<App />);
});

it("renders correctly", () => {
  const tree = renderer.create(<App></App>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("Doesn't show a dog image when the page renders");

it("Shows a dog image when the 'Next Dog' button is pressed");

it("unmounts the dog image if the user 💔s the dog");
it("unmounts the dog image if the user ❤️s the dog");

it("Renders the matched dog when the loved dog is matched");

it("Makes sure the same name is not used twice");

it("When matched, the previous dog image is rendered");
