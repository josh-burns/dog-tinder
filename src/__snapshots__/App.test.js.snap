// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-gKsewC jiYbnw"
>
  <div>
    <div
      className="sc-jSgupP gPImzZ"
    >
      <div>
        <image
          className="sc-hKgILt fNbktM"
        >
          <a
            href="/"
          >
            <img
              alt="logo"
              src="fetchr.png"
            />
          </a>
        </image>
        Find your canine Match on Fetchr!
        <br />
        <br />
        Choose ‚ù§Ô∏è or üíî and if they ‚ù§Ô∏è you...
        <br />
        <br />
        It's a match!
        <br />
        <br />
        <div
          className="sc-eCssSg cRgFNj"
          onClick={[Function]}
        >
          <h2>
            Next Dog 
          </h2>
        </div>
        <div />
      </div>
    </div>
  </div>
</div>
`;
